### ISSUE TITLE
Dask config fails to load

### ISSUE THREAD

**Issue by thibaut-kobold** [dask/dask#8799]
Hi,
I encounter an issue where I can succesfully install Dask from conda, but fails to import it later. I attached below a minimal example where:
- I highlight at the start conda/python versions
- I create and activate a blank conda environment with python 3.9
- install only dask with `conda install dask`
- run python and try to import dask, with no success.

Any idea of what is happening? Thanks in advance for your help.
I am on a macbook pro with intel processors.

```
Last login: [redacted] on ttys001
(base) thibautkobold@[redacted] ~ % conda --version
conda 4.11.0
(base) thibautkobold@[redacted] ~ % which python
[redacted]/anaconda3/bin/python
(base) thibautkobold@[redacted] ~ % python --version
Python 3.9.7
(base) thibautkobold@[redacted] ~ % conda create --name dask-env-py39 python=3.9
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: [redacted]/anaconda3/envs/dask-env-py39

  added / updated specs:
    - python=3.9


The following NEW packages will be INSTALLED:

  ca-certificates    pkgs/main/osx-64::ca-certificates-2022.2.1-hecd8cb5_0
  certifi            pkgs/main/osx-64::certifi-2021.10.8-py39hecd8cb5_2
  libcxx             pkgs/main/osx-64::libcxx-12.0.0-h2f01273_0
  libffi             pkgs/main/osx-64::libffi-3.3-hb1e8313_2
  ncurses            pkgs/main/osx-64::ncurses-6.3-hca72f7f_2
  openssl            pkgs/main/osx-64::openssl-1.1.1m-hca72f7f_0
  pip                pkgs/main/osx-64::pip-21.2.4-py39hecd8cb5_0
  python             pkgs/main/osx-64::python-3.9.7-h88f2d9e_1
  readline           pkgs/main/osx-64::readline-8.1.2-hca72f7f_1
  setuptools         pkgs/main/osx-64::setuptools-58.0.4-py39hecd8cb5_0
  sqlite             pkgs/main/osx-64::sqlite-3.37.2-h707629a_0
  tk                 pkgs/main/osx-64::tk-8.6.11-h7bc2e8c_0
  tzdata             pkgs/main/noarch::tzdata-2021e-hda174b7_0
  wheel              pkgs/main/noarch::wheel-0.37.1-pyhd3eb1b0_0
  xz                 pkgs/main/osx-64::xz-5.2.5-h1de35cc_0
  zlib               pkgs/main/osx-64::zlib-1.2.11-h4dc903c_4


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate dask-env-py39
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) thibautkobold@[redacted] ~ % conda activate dask-env-py39
(dask-env-py39) thibautkobold@[redacted] ~ % which python
[redacted]/anaconda3/envs/dask-env-py39/bin/python
(dask-env-py39) thibautkobold@[redacted] ~ % python --version
Python 3.9.7
(dask-env-py39) thibautkobold@[redacted] ~ % conda install dask
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: [redacted]/anaconda3/envs/dask-env-py39

  added / updated specs:
    - dask


The following NEW packages will be INSTALLED:

  blas               pkgs/main/osx-64::blas-1.0-mkl
  bokeh              pkgs/main/osx-64::bokeh-2.4.2-py39hecd8cb5_0
  bottleneck         pkgs/main/osx-64::bottleneck-1.3.2-py39he3068b8_1
  click              pkgs/main/osx-64::click-8.0.4-py39hecd8cb5_0
  cloudpickle        pkgs/main/noarch::cloudpickle-2.0.0-pyhd3eb1b0_0
  cytoolz            pkgs/main/osx-64::cytoolz-0.11.0-py39h9ed2024_0
  dask               pkgs/main/noarch::dask-2021.10.0-pyhd3eb1b0_0
  dask-core          pkgs/main/noarch::dask-core-2021.10.0-pyhd3eb1b0_0
  distributed        pkgs/main/osx-64::distributed-2021.10.0-py39hecd8cb5_0
  freetype           pkgs/main/osx-64::freetype-2.11.0-hd8bbffd_0
  fsspec             pkgs/main/noarch::fsspec-2022.1.0-pyhd3eb1b0_0
  giflib             pkgs/main/osx-64::giflib-5.2.1-haf1e3a3_0
  heapdict           pkgs/main/noarch::heapdict-1.0.1-pyhd3eb1b0_0
  intel-openmp       pkgs/main/osx-64::intel-openmp-2021.4.0-hecd8cb5_3538
  jinja2             pkgs/main/noarch::jinja2-3.0.2-pyhd3eb1b0_0
  jpeg               pkgs/main/osx-64::jpeg-9d-h9ed2024_0
  lcms2              pkgs/main/osx-64::lcms2-2.12-hf1fd2bf_0
  libpng             pkgs/main/osx-64::libpng-1.6.37-ha441bb4_0
  libtiff            pkgs/main/osx-64::libtiff-4.2.0-h87d7836_0
  libwebp            pkgs/main/osx-64::libwebp-1.2.2-h56c3ce4_0
  libwebp-base       pkgs/main/osx-64::libwebp-base-1.2.2-hca72f7f_0
  locket             pkgs/main/osx-64::locket-0.2.1-py39hecd8cb5_1
  lz4-c              pkgs/main/osx-64::lz4-c-1.9.3-h23ab428_1
  markupsafe         pkgs/main/osx-64::markupsafe-2.0.1-py39h9ed2024_0
  mkl                pkgs/main/osx-64::mkl-2021.4.0-hecd8cb5_637
  mkl-service        pkgs/main/osx-64::mkl-service-2.4.0-py39h9ed2024_0
  mkl_fft            pkgs/main/osx-64::mkl_fft-1.3.1-py39h4ab4a9b_0
  mkl_random         pkgs/main/osx-64::mkl_random-1.2.2-py39hb2f4e1b_0
  msgpack-python     pkgs/main/osx-64::msgpack-python-1.0.2-py39hf7b0b51_1
  numexpr            pkgs/main/osx-64::numexpr-2.8.1-py39h2e5f0a9_0
  numpy              pkgs/main/osx-64::numpy-1.21.2-py39h4b4dc7a_0
  numpy-base         pkgs/main/osx-64::numpy-base-1.21.2-py39he0bd621_0
  packaging          pkgs/main/noarch::packaging-21.3-pyhd3eb1b0_0
  pandas             pkgs/main/osx-64::pandas-1.4.1-py39he9d5cce_0
  partd              pkgs/main/noarch::partd-1.2.0-pyhd3eb1b0_0
  pillow             pkgs/main/osx-64::pillow-9.0.1-py39hde71d04_0
  psutil             pkgs/main/osx-64::psutil-5.8.0-py39h9ed2024_1
  pyparsing          pkgs/main/noarch::pyparsing-3.0.4-pyhd3eb1b0_0
  python-dateutil    pkgs/main/noarch::python-dateutil-2.8.2-pyhd3eb1b0_0
  pytz               pkgs/main/noarch::pytz-2021.3-pyhd3eb1b0_0
  pyyaml             pkgs/main/osx-64::pyyaml-6.0-py39hca72f7f_1
  six                pkgs/main/noarch::six-1.16.0-pyhd3eb1b0_1
  sortedcontainers   pkgs/main/noarch::sortedcontainers-2.4.0-pyhd3eb1b0_0
  tblib              pkgs/main/noarch::tblib-1.7.0-pyhd3eb1b0_0
  toolz              pkgs/main/noarch::toolz-0.11.2-pyhd3eb1b0_0
  tornado            pkgs/main/osx-64::tornado-6.1-py39h9ed2024_0
  typing_extensions  pkgs/main/noarch::typing_extensions-3.10.0.2-pyh06a4308_0
  yaml               pkgs/main/osx-64::yaml-0.2.5-haf1e3a3_0
  zict               pkgs/main/noarch::zict-2.0.0-pyhd3eb1b0_0
  zstd               pkgs/main/osx-64::zstd-1.4.9-h322a384_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(dask-env-py39) thibautkobold@[redacted] ~ % python
Python 3.9.7 (default, Sep 16 2021, 08:50:36) 
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/__init__.py", line 1, in <module>
    from . import config, datasets
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/config.py", line 653, in <module>
    refresh()
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/config.py", line 431, in refresh
    update(config, collect(**kwargs))
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/config.py", line 399, in collect
    return merge(*configs)
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/config.py", line 123, in merge
    update(result, d)
  File "[redacted]/anaconda3/envs/dask-env-py39/lib/python3.9/site-packages/dask/config.py", line 91, in update
    for k, v in new.items():
AttributeError: 'str' object has no attribute 'items'
>>> 

```

**Comment by jcrist** [comment]
I'd guess you have a malformed config file (containing only a string, not a map) somewhere on your system. Dask loads yaml files from several places on import:

- `~/.config/dask/`
- `/etc/dask/`
- `{prefix}/etc/dask/` where `prefix` is the root of your python environment

Unless you or an administrator put a config file in one of these places though, I'd expect a clean install to load no config files.

**Comment by jcrist** [comment]
That said, we should provide a better error message if a config file doesn't load properly or is malformed in this way.

**Comment by ncclementi** [comment]
@thibaut-kobold Jim beat me to it. I doubled check and followed the steps you mentioned and I had no problem importing dask. But something else I'm noticing is that the version of dask that gets installed in your environment is `2021.10`, which is not the latest `2022.2.1`. Not sure if this is what you desired, but if you want the latest version I would recommend adding `conda-forge` as your default channel.

**Comment by thibaut-kobold** [comment]
Thank you for your help @jcrist and @ncclementi, this is helpful. I will check the config files mentioned. Dask version is not the issue (I reproduced the error with both 2021 and 2022 versions).
As the error is not reproducible on other machines, feel free to close this issue.

**Comment by thibaut-kobold** [comment]
@jcrist : you were right! A config file in ~/.config/dask/ was causing the issue. Many thanks!

**Comment by jcrist** [comment]
Glad to hear it! I've pushed a fix up at #8801 that should give a better error message in this case. Should hopefully help the next user in this situation debug things easier.
